app/lib/rocketchat/services/shareExtension.ts

```diff
@@ -0,0 +1,93 @@
+import { Q } from '@nozbe/watermelondb';
+
+import { shareSetSettings, shareSelectServer, shareSetUser } from '../../../actions/share';
+import SSLPinning from '../../../utils/sslPinning';
+import log from '../../../utils/log';
+import { IShareServer, IShareUser } from '../../../reducers/share';
+import UserPreferences from '../../userPreferences';
+import database from '../../database';
+import RocketChat from '../rocketchat';
+import { encryptionInit } from '../../../actions/encryption';
+import { store } from '../../auxStore';
+import sdk from './sdk';
+import { IRocketChat } from '../../../definitions';
+
+export async function shareExtensionInit(this: IRocketChat, server: string) {
+	database.setShareDB(server);
+
+	try {
+		const certificate = await UserPreferences.getStringAsync(`${RocketChat.CERTIFICATE_KEY}-${server}`);
+		if (SSLPinning && certificate) {
+			await SSLPinning.setCertificate(certificate, server);
+		}
+	} catch {
+		// Do nothing
+	}
+
+	this.shareSDK = sdk.disconnect();
```

Comment:
I think there are other better ways to do this implementation without using this.